ddtrace/contrib/tornado/__init__.py

```diff
@@ -0,0 +1,73 @@
+"""
+The Tornado integration traces all ``RequestHandler`` defined in a Tornado web application.
+Auto instrumentation is available using the ``trace_app`` function as follows::
+
+    import tornado.web
+    import tornado.gen
+    import tornado.ioloop
+
+    from ddtrace import tracer
+    from ddtrace.contrib.tornado import trace_app
+
+    # create your handlers
+    class MainHandler(tornado.web.RequestHandler):
+        @tornado.gen.coroutine
+        def get(self):
+            self.write("Hello, world")
+
+    # create your application
+    app = tornado.web.Application([
+        (r'/', MainHandler),
+    ])
+
+    # trace your application before the execution
+    trace_app(app, tracer, service='tornado-site')
+
+    # and run it as usual
+    app.listen(8888)
+    tornado.ioloop.IOLoop.current().start()
+
+When a ``RequestHandler`` is hit, a request span is automatically created and attached
+to the current ``request`` object, so that it can be used in the application code::
+
+    class MainHandler(tornado.web.RequestHandler):
+        @tornado.gen.coroutine
+        def get(self):
+            ctx = getattr(self.request, 'datadog_context')
+            # do something with the tracing Context
+
+If you want to trace other part of your application, you can use both the ``Tracer.wrap()``
+decorator and the ``Tracer.trace()`` method::
+
+    class MainHandler(tornado.web.RequestHandler):
+        @tornado.gen.coroutine
+        def get(self):
+            yield self.notify()
+            with tracer.trace('tornado.post_notify') as span:
+                # do more work
+
+        @tracer.wrap('tornado.notify', service='tornado-notification')
```

Comment:
This will time the generation of the future object (essentially 0s) and not how long the `notify` method takes to "do something". If instead the put `@trace.wrap` is on the other side of `@tornado.gen.coroutine` it'll time things correctly, but not be assigned a parent span id.